lineman/app/components/thread_page/proposal_positions_panel/proposal_positions_panel.coffee

```diff
@@ -6,7 +6,8 @@ angular.module('loomioApp').directive 'proposalPositionsPanel', ->
   controller: ($scope, Records, CurrentUser, ModalService, VoteForm) ->
 
     $scope.changeVote = ->
-      ModalService.open VoteForm, proposal: -> $scope.proposal
+      ModalService.open VoteForm, 
+        proposal: -> $scope.proposal.clone()
```

Comment:
FYI: In both those examples you need to wrap the resolve value in a function or I don't think they'll work.

We could add a key to the modal/form: "requiredArguments" and list resolves that must be supplied or raise an exception.